lib/db/mysql.js

```diff
@@ -916,15 +916,23 @@ module.exports = function (log, error) {
 
   // Update : emails
   // Values : uid = $1, email = $2
-  var SET_PRIMARY_EMAIL = 'CALL setPrimaryEmail_3(?, ?)'
+  var SET_PRIMARY_EMAIL = 'CALL setPrimaryEmail_4(?, ?)'
   MySql.prototype.setPrimaryEmail = function (uid, email) {
-    return this.write(
-      SET_PRIMARY_EMAIL,
-      [
-        uid,
-        email
-      ]
-    )
+    return this.write(SET_PRIMARY_EMAIL, [uid, email])
+      .then(() => {
+        return {}
+      })
+      .catch(err => {
+        if (err.errno === error.duplicate().errno) {
```

Comment:
Looks like we could benefit from the auth-server's approach of giving our errono's easy named constants, but that's beyond the scope of this PR :-)

Reply:
Good idea! Created https://github.com/mozilla/fxa-auth-db-mysql/issues/447, so we don't lose track.